src/crimson/os/seastore/seastore_types.h

```diff
@@ -37,26 +37,121 @@ struct seastore_meta_t {
 
```

Comment:
If I'm understanding this correctly, there are now three types in play:
1. device_segment_t: denotes a specific segment, seastore-wide.
2. device_segment_id_t: denotes a specific segment within an already specified device.  Should really only be used when translating operations on a device_segment_t to a specific device
3. segment_id_t: same as device_segment_id_t

I don't understand the distinction between segment_id_t and device_segment_id_t.  I also think that device_segment_t should be renamed to segment_id_t since that would change the fewest users and because it identifies a segment.  segment_id_t/device_segment_id_t would then be device_segment_idx_t.

MAX_SEG_ID, NULL_SEG_ID, RECORD_REL_SEG_ID, etc should have the type used in paddr_t.

Reply:
> 3\. segment_id_t: same as device_segment_id_t

My intention was to use segment_id_t to represent segment IDs without deviced id.

Reply:
Should we distinguish segment ids with device ids from those without device ids?

Reply:
device_segment_id_t seems to also represent a segment id without a device.

Reply:
I think the context of `segment_id_t`/`device_segment_id_t` is from https://github.com/ceph/ceph/pull/41803#discussion_r683179785

Reply:
I think that's the difference between device_segment_t and device_segment_id_t/segment_id_t. I'm confused because while device_segment_t denotes a device/segment combination, both device_segment_id_t and segment_id_t seem to denote a segment without a device.